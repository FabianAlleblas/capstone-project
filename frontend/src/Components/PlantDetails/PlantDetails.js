import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import defaultPlantImage from '../../assets/plant-images/default_plant.jpg'

PlantDetails.propTypes = {
  plant: PropTypes.object.isRequired,
}

export default function PlantDetails({ plant }) {
  return (
    <>
      <ImageFrame src={plant.image ?? defaultPlantImage} />
      <TextContainer>
        <PlantName>{plant.species}</PlantName>
        <PlantInfo>{plant.specialInfo}</PlantInfo>
      </TextContainer>
    </>
  )
}

const ImageFrame = styled.div`
  background-image: url(${(props) => props.src});
  background-position: center;
  background-size: cover, contain;
  border-radius: 50%;
  height: 200px;
  width: 200px;
`

const TextContainer = styled.div`
  line-height: 1.2;
  text-align: center;
`

const PlantName = styled.h2`
  color: var(--primary-plant-font-color);
  font-size: 1.4rem;
  font-weight: 700;
`

const PlantInfo = styled.h3`
  color: var(--secondary-plant-font-color);
  font-size: 1.4rem;
  font-weight: 400;
`
