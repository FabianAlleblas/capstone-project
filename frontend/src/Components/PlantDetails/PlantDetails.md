```js
<PlantDetails
  plant={{
    species: 'Monstera deliciosa',
    info: 'variegated',
  }}
/>
```
