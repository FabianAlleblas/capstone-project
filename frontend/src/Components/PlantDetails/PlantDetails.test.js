import { render } from '@testing-library/react'
import 'jest-styled-components'
import PlantDetails from './PlantDetails'

describe('PlantDetailCard', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PlantDetails
        plant={{
          species: 'Monstera',
          specialInfo: 'variegated',
        }}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows the correct name and info', () => {
    const { getByText } = render(
      <PlantDetails
        plant={{
          species: 'Monstera',
          specialInfo: 'variegated',
        }}
      />
    )

    expect(getByText('Monstera')).toBeInTheDocument()
    expect(getByText('variegated')).toBeInTheDocument()
  })
})
