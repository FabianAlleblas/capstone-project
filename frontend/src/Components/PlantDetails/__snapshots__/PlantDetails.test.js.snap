// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantDetailCard renders correctly 1`] = `
.c0 {
  background-image: url(default_plant.jpg);
  background-position: center;
  background-size: cover,contain;
  border-radius: 50%;
  height: 200px;
  width: 200px;
}

<div
  class="c0"
  src="default_plant.jpg"
/>
`;
